src/types/resolveStatements.ts

```diff
@@ -486,11 +498,12 @@ function processStatements(
                     let catchCtx = sctx;
 
                     // Process catchName variable for exit code
-                    checkVariableExists(initialCtx, s.catchName);
+                    checkVariableExists(ctx, initialSctx, s.catchName);
```

Comment:
We could try refactor this, although I'm not sure how useful this is going to be since we are going to replace the existing typechecker with a new one

Reply:
Agreed. I think there is no need to refactor.